includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+use \EDD\Admin\Extensions\ProductData;
+
+class ExtensionsAPI {
+
+	/**
+	 * The ProductData class.
+	 *
+	 * @var \EDD\Admin\Extensions\ProductData
+	 */
+	private $product_data;
+
+	public function __construct() {
+		$this->product_data = new ProductData();
+	}
+
+	/**
+	 * Gets the product data from the EDD Products API.
+	 *
+	 * @since 2.11.x
+	 * @param array $body    The body for the API request.
+	 * @param int   $item_id The product ID, if querying a single product.
+	 * @return false|array|\ProductData
+	 */
+	public function get_product_data( $body = array(), $item_id = false ) {
+		if ( empty( $body ) ) {
+			if ( empty( $item_id ) ) {
+				return false;
+			}
+			$body = $this->get_api_body( $item_id );
+		}
+		$key = $this->array_key_first( $body );
+		// The option name is created from the first key/value pair of the API "body".
+		$option_name = sanitize_key( "edd_extension_{$key}_{$body[ $key ]}_data" );
+		$option      = get_option( $option_name );
+		$is_stale    = ! empty( $option['timeout'] ) && time() > $option['timeout'];
+
+		// If the data is "fresh" and what we want exists, return it.
+		if ( ! $is_stale ) {
+			if ( $item_id && ! empty( $option[ $item_id ] ) ) {
+				return is_array( $option[ $item_id ] ) ? $this->product_data->fromArray( $option[ $item_id ] ) : $option[ $item_id ];
```

Comment:
@ashleyfae This and line 70 do not necessarily have to stay this way--they are in so that anyone testing this PR won't have explosions over the change from the array to the `ProductData` object. Once I can safely assume that anyone would be getting fresh/new options created, this should just be able to be:

```php
return $option[ $item_id ];
```

If I do that, then I could remove the product data property from the class and constructor and just instantiate the product data class down in the `get_item_data` method, I think, and skip the whole array to object conversion, I think?

Reply:
I think this is outdated after https://github.com/awesomemotive/easy-digital-downloads/pull/8993#discussion_r766967065 ?